packages/slice-machine/src/modules/availableCustomTypes/index.ts

```diff
@@ -208,23 +211,48 @@ export const availableCustomTypesReducer: Reducer<
     }
 
     case getType(deleteCustomTypeCreator.success): {
-      return omit(state, action.payload.customTypeId);
+      const ct = state[action.payload.customTypeId];
+      if (isSyncedCustomType(ct)) {
+        const deletedCT: DeletedFrontEndCustomType = omit(ct, "local");
+        return {
+          ...state,
+          [ct.local.id]: deletedCT,
+        };
+      }
+      if (isNewCustomType(ct)) {
+        return omit(state, action.payload.customTypeId);
+      }
+
+      return state;
```

Comment:
[OPT] Do you think this case deserves a comment?

Reply:
I'd say no, the types are explicit enough